### PR TITLE
gha: tests: Add kata-deploy functional tests -- Part 1

### PR DESCRIPTION
**Pull request by fidencio**
This first part does:
* Splits gha-run.sh from the `integration/kubernetes`  dir, so it can be reused for the `functional/kata-deploy`  tests that will come in the future
* Add the aks / tdx tests, as those are the ones I will be touching
  * sev / snp tests are also welcome but I'll leave this to AMD to play with
* Add a basic placeholder for the gha-run.sh, which will be used by the `functional/kata-deploy` tests.

### CODE REVIEW

**Review by sprt — APPROVED** [review]
LGTM

File: .github/workflows/run-kata-deploy-tests-on-aks.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: CI | Run kubernetes tests on AKS
```
Comment (wainersm) [inline .github/workflows/run-kata-deploy-tests-on-aks.yaml:1]: rename the job?

Comment (fidencio) [reply .github/workflows/run-kata-deploy-tests-on-aks.yaml]: Just did, thanks!

**Review by jepio — APPROVED** [review]

**Review by wainersm — APPROVED** [review]

**Comment by fidencio** [comment]
I'm merging this one even with the power CI not running, as this is not related to jenkins at all.

**Pull request merged**